### ISSUE TITLE
webpack@5 output.libraryTarget: "module" and html-webpack-plugin cause compile error

### ISSUE THREAD

**Issue by mizchi** [jantimon/html-webpack-plugin#1492]
## Expected behaviour

Generate html

## Current behaviour

```
ERROR in   Error: /Users/mizchi/plg/try-microfront/src/index.html:10
  /******/ var __webpack_modules__ = ({
           ^^^
  SyntaxError: Unexpected token 'var'
  
  - index.js:310 HtmlWebpackPlugin.evaluateCompilationResult
    [try-microfront]/[html-webpack-plugin]/index.js:310:22
  
  - index.js:230 
    [try-microfront]/[html-webpack-plugin]/index.js:230:22
  
  - task_queues.js:93 processTicksAndRejections
    internal/process/task_queues.js:93:5
  
  - async Promise.all
  
  - async Promise.all
```

## Environment

- Node.js v14.7.0 darwin 19.5.0
- npm@6.14.7
- webpack@5.0.0-beta.24
- html-webpack-plugin@4.3.0

### Config

```js
const HtmlPlugin = require("html-webpack-plugin");
const path = require("path");

module.exports = {
  output: {
    libraryTarget: "module"
  },
  plugins: [
    new HtmlPlugin({
      template: path.join(__dirname, "src/index.html")
    })
  ],
  experiments: {
    outputModule: true
  }
}
```

**Comment by miyaokamarina** [comment]
Same with `experiments.outputModule = true, output.module = true`.

node 14.13.0
webpack 5.1.0
html-webpack-plugin 4.5.0

**Comment by jantimon** [comment]
A workaround might be to use the `templateContent` option for now - has anyone an idea how we could fix this issue?

**Comment by andrew-t-james-hm** [comment]
➕ 1

**Comment by jantimon** [comment]
Can you please try if the new html-webpack-plugin@next version works better for you?

**Comment by andrew-t-james-hm** [comment]
@jantimon That worked, thanks for the heads up on the alpha-version.

**Comment by Zaynex** [comment]
still fail.

```
const path = require("path")
const HtmlWebpackPlugin = require('html-webpack-plugin'); // 通过 npm 安装
const { CleanWebpackPlugin } = require("clean-webpack-plugin")
module.exports = {
  mode: "development",
  entry: "./src/index.js",
  output: {
    module: true,
    path: path.resolve(__dirname, 'dist'),
  },
  experiments: {
    outputModule: true
  },
  devtool: 'inline-source-map',
  devServer: {
    contentBase: "./dist"
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        exclude: /node_modules/,
        use: ["babel-loader"]
      }
    ]
  },
  plugins: [
    new CleanWebpackPlugin({ cleanStaleWebpackAssets: false}),
    new HtmlWebpackPlugin({template: path.resolve(__dirname, "src", "index.html")})
  ]
}
```
```
"clean-webpack-plugin": "^3.0.0",
"html-webpack-plugin": "^5.0.0-alpha.15",
"webpack": "^5.10.0",
"webpack-cli": "^4.2.0",
"webpack-dev-server": "^3.11.0"
```

**Comment by Dylan0916** [comment]
The same happened to me
`"html-webpack-plugin": "^5.0.0-alpha.15"`

**Comment by jantimon** [comment]
The error is fixed in `html-webpack-plugin@5.0.0-alpha.16` however the inject feature will not add the `module` attribute to the script tag automatically

**Comment by xoxefdp** [comment]
> The error is fixed in `html-webpack-plugin@5.0.0-alpha.16` however the inject feature will not add the `module` attribute to the script tag automatically

you mean type="module" to the bundle in html?
